### PR TITLE
WW-5381 Introduce extension point for MethodAccessor

### PR DESCRIPTION
**Pull request by kusalk**
WW-5381
--
Fairly straight forward again but see my comment below

### CODE REVIEW

File: core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java [hunk 1]
```diff
@@ -87,12 +92,6 @@ protected void setContainer(Container container) throws ClassNotFoundException {
             OgnlRuntime.setPropertyAccessor(cls, container.getInstance(PropertyAccessor.class, name));
         }
 
-        names = container.getInstanceNames(MethodAccessor.class);
```
Comment (kusalk) [inline core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java:90]: Now there is one drawback here - applications lose the ability to specify custom method accessors for different types. They will now instead need to use the extension point and do the delegation within this "global" method accessor. I'm not sure there's a way to retain this capability and make it compatible with the current extension point system.

**Comment by kusalk** [comment]
(SonarCloud report is comparing against wrong base branch)

File: core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java (hunk 1 above)
Comment (lukaszlenart) [reply core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java:90]: Hmm... so maybe just mark this functionality as `@deprecated` or remove - once someone starts complain we can re-think how to put back such functionality.

**Review by lukaszlenart — APPROVED** [review]

File: core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java (hunk 1 above)
Comment (kusalk) [reply core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java:90]: Hmm actually just had another think and we should be able to retain backwards compatibility, the code will be a little ugly though - let me give it a shot

Comment (kusalk) [reply core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java:90]: I think we are good now :) Hopefully the code makes sense, added some JavaDoc

Comment (lukaszlenart) [reply core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java:90]: Looks good to me, thanks!

**Review by lukaszlenart — APPROVED** [review]

**Pull request merged**